[Masonry] First Possible Solution of Masonry Component (#27169)
mui-org/material-ui: docs/src/pages/components/masonry/masonry.md

---
title: React Masonry component
components: Masonry, MasonryItem
githubLabel: 'component: Masonry'
---

# Masonry

<p class="description">Masonry lays out contents of different sizes as blocks of the same width and variable height with configurable gaps.</p>

Masonry maintains a list of content blocks with a consistent width but variable height. The contents are ordered by row. If a row is already filled with the specified number of columns, the next item starts another row, and it is added to the shortest column.

> {{"component": "modules/components/ComponentLinkHeader.js"}}

Reviewer: We won't have it in the design kits, so no need to link them

```suggestion
{{"component": "modules/components/ComponentLinkHeader.js", "design": true}}
```

---
title: React Masonry component
components: Masonry
githubLabel: 'component: Masonry'
---

# Masonry

> <p class="description">With horizontal ordering and column stacking, Masonry lays out content of different sizes without leaving gaps.</p>

Reviewer 2: ```suggestion
<p class="description">With horizontal ordering and column stacking, Masonry lays out content of different sizes with configurable gaps.</p>
```

> Masonry maintains a list of content blocks with a consistent width but variable height. The contents are ordered from left to right and top to bottom. If a row is already filled with the specified number of columns, the next item starts another row starts, and it is added to the shortest column.

Reviewer 2: ```suggestion
Masonry maintains a list of content blocks with a consistent width but variable height. The contents are ordered from left to right and top to bottom. If a row is already filled with the specified number of columns, the next item starts another row, and it is added to the shortest column.
```

{{"component": "modules/components/ComponentLinkHeader.js"}}

## Basic Masonry

> {{"demo": "pages/components/masonry/BasicMasonry.js", "bg": true}}

Reviewer 2: Let's try to keep the demo short, so that it can be previewed. Then we wouldn't need the code snippet below
